### PR TITLE
DDS export of vss generated IDL fails Fix

### PR DESCRIPTION
**Pull request by AkhilTThomas**
Added C,Python and IDL keyword crosss checking and appended underscore in front of keywords
Ex: MAP becomes _MAP
Also enums now exist in their own namespace under a module wrapper, the modules name is the enum name+_M
Ex:
module LowVoltageSystemState_M` { enum LowVoltageSystemStateValues{UNDEFINED,LOCK,OFF,ACC,ON,START}; }; struct LowVoltageSystemState { LowVoltageSystemState_M::LowVoltageSystemStateValues value; //const string type ="sensor"; //const string  #description="State of the supply voltage of the control units (usually 12V)."; };
#185

### CODE REVIEW

**Comment by SebastianSchildt** [comment]
(note: CI test failure is to be expected, because here the fix is changing the idl output, and the CI simply tests an example file against an expected known output)

**Comment by SebastianSchildt** [comment]
Not an DDS-IDL expert, but just with my VSS head on, I appreciate the progress of having a valid/legal DDS output from VSS 😁 . @nayakned since I happen to know you are using DDS with VSS, can you check whether this variant of DDS-IDL mapping would work with your use cases/models, and give feedback?

**Review by erikbosch — APPROVED** [review]

**Comment by erikbosch** [comment]
@AkhilTThomas did you close it by mistake? It is a PR so to be included in master it must be merged. I marked this PR as approved, but wanted input from others before clicking the merge button.

If it was closed by mistake you or me can reopen it.

**Comment by AkhilTThomas** [comment]
@erikbosch I have opened it.

**Comment by nayakned** [comment]
> Not an DDS-IDL expert, but just with my VSS head on, I appreciate the progress of having a valid/legal DDS output from VSS grin . @nayakned since I happen to know you are using DDS with VSS, can you check whether this variant of DDS-IDL mapping would work with your use cases/models, and give feedback?

Looks ok to me. I think our usecases should (still) work. However, this doesnt completely solve #185 I think. Since the problem with redefined enums would remains.

**Comment by AkhilTThomas** [comment]
> Looks ok to me. I think our usecases should (still) work. However, this doesnt completely solve #185 I think. Since the problem with redefined enums would remains.

The enums are now encased in their own module block and accessed via this module scope. This avoids the conflict

**Comment by nayakned** [comment]
> The enums are now encased in their own module block and accessed via this module scope. This avoids the conflict

Ah. had missed that part. Thanks for the clarification.

**Comment by erikbosch** [comment]
Merging it to allow some other PRs to be updated.

**Pull request merged**
